models/ed/R/SAS.ED2.R

```diff
@@ -0,0 +1,502 @@
+                    rh_lloyd_1=308.56, rh_lloyd_2=1/56.02, rh_lloyd_3=227.15
+                    ) {
+  if(!decomp_scheme %in% 0:4) stop("Invalid decomp_scheme")
+  # create a directory for the initialization files
+  dir.create(outdir, recursive=T, showWarnings=F)
+  
+  #---------------------------------------
+  # Setting up some specifics that vary by site (like soil depth)
+  #---------------------------------------
+  #Set directories
+  # dat.dir    <- dir.analy
+  ann.files  <- dir(dir.analy, "-Y-") #yearly files only  
+  
+  #Get time window
+  # Note: Need to make this more flexible to get the thing after "Y"
+  yrind <- which(strsplit(ann.files,"-")[[1]] == "Y")
+  yeara  <- as.numeric(strsplit(ann.files,"-")[[1]][yrind+1]) #first year
+  yearz  <- as.numeric(strsplit(ann.files,"-")[[length(ann.files)]][yrind+1]) #last full year
+  yrs    <- seq(yeara+1, yearz, by=blckyr) # The years we're going to use as time steps for the demography
+  nsteps <- length(yrs) # The number of blocks = the number steps we'll have
+  
+  # Need to get the layers being used for calculating temp & moist
+  # Note: In ED there's a pain in the butt way of doing this with the energy, but we're going to approximate
+  # slz  <- c(-5.50, -4.50, -2.17, -1.50, -1.10, -0.80, -0.60, -0.45, -0.30, -0.20, -0.12, -0.06)
+  # dslz <- c(1.00,   2.33,  0.67,  0.40,  0.30,  0.20,  0.15,  0.15,  0.10,  0.08,  0.06,  0.06)
+  nc.temp <- ncdf4::nc_open(file.path(dir.analy, ann.files[1]))
+  slz <- ncdf4::ncvar_get(nc.temp, "SLZ")
+  ncdf4::nc_close(nc.temp)
+  
+  dslz <- vector(length=length(slz))
+  dslz[length(dslz)] <- 0-slz[length(dslz)]
+  
+  for(i in 1:(length(dslz)-1)){
+    dslz[i] <- slz[i+1] - slz[i]    
+  }
+  
+  nsoil=which(slz >= kh_active_depth-1e-3)  # Maximum depth for avg. temperature and moisture; adding a fudge factor bc it's being weird
+  # nsoil=length(slz)
+  #---------------------------------------
+  
+  #---------------------------------------
+  # First loop over analy files (faster than histo) to aggregate initial 
+  # 	.css and .pss files for each site
+  #---------------------------------------
+  #create an emtpy storage for the patch info
+  pss.big <- matrix(nrow=length(yrs),ncol=13) # save every X yrs according to chunks specified above
+  colnames(pss.big) <- c("time","patch","trk","age","area","water","fsc","stsc","stsl",
+                         "ssc","psc","msn","fsn")
+
+  #---------------------------------------
+  # Finding the mean soil temp & moisture
+  # NOTE:  I've been plyaing around with finding the best temp & soil moisture to initialize things
+  #        with; if using the means from the spin met cycle work best, insert them here
+  # This will also be necessary for helping update disturbance parameter
+  #---------------------------------------
+  slmsts <- calc.slmsts(slxsand, slxclay)
+  slpots <- calc.slpots(slxsand, slxclay)
+  slbs   <- calc.slbs(slxsand, slxclay)
+  soilcp <- calc.soilcp(slmsts, slpots, slbs)
```

Comment:
* These functions don't seem to be defined anywhere
* They seem redundant with the soil_params function https://github.com/PecanProject/pecan/blob/develop/modules/data.land/R/soil_utils.R#L26

Reply:
I didn't know about the soil_params function, so I'll check that out.  These are from ED's internal routines, so if they aren't consistent with soil_params, would it be better to add a new script with ED soil functions or tack it on here?

Reply:
FWIW soil_params is based on some R code that Marcos wrote specifically for ED2. That doesn't guarantee the params will be exactly the same, but I'm hopeful that's the case